Lib/test/test_xml_etree.py

```diff
@@ -2693,6 +2694,41 @@ def __eq__(self, o):
         e.extend([ET.Element('bar')])
         self.assertRaises(ValueError, e.remove, X('baz'))
 
+    def test_remove_with_clear_children(self):
+        # See: https://github.com/python/cpython/issues/126033
+
+        class X(ET.Element):
+            def __eq__(self, o):
+                root.clear()
+                return False
+
+        for foo_type, rem_type in [(X, ET.Element), (ET.Element, X)]:
+            with self.subTest(foo_type=foo_type, rem_type=rem_type):
+                root = ET.Element('.')
+                root.extend([foo_type('foo'), rem_type('bar')])
+                self.assertRaises(ValueError, root.remove, rem_type('baz'))
+
+    def test_remove_with_mutate_root(self):
+        # See: https://github.com/python/cpython/issues/126033
+
+        first_element = ET.Element('foo')
+
+        class X(ET.Element):
+            def __eq__(self, o):
+                # Remove the first element so that the list size changes.
+                # This causes an infinite recursion error in the Python
+                # implementation, but we do not really care about it.
+                #
+                # Depending on whether the first element is or is not
+                root.remove(first_element)
+                return False
+
+        for bar_type, rem_type in [(X, ET.Element), (ET.Element, X), (X, X)]:
```

Comment:
What is the difference between these cases? Why testing only one case is not enough?

Reply:
I wanted to be sure that there won't be a symmetry issue. Do you think it's too much?